### PR TITLE
added a greedy method

### PR DESCRIPTION
**Pull request by HN026**
### Describe your change:

* [ . ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: greedy_methods/interval_scheduling.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Interval Scheduling: https://en.wikipedia.org/wiki/Interval_scheduling
```
Comment (tianyizheng02) [inline greedy_methods/interval_scheduling.py:1]: "Interval scheduling" is the name of the problem, not the algorithm. What is the name of the algorithm?

Comment (tianyizheng02) [inline greedy_methods/interval_scheduling.py:1]: Also, please provide an explanation of the algorithm: what problem it solves (interval scheduling, in this case), how it works, its time complexity, etc.

File: greedy_methods/interval_scheduling.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def interval_scheduling(intervals):
```
Comment (tianyizheng02) [inline greedy_methods/interval_scheduling.py:4]: Rename this function to match the name of the _algorithm,_ not the problem.

Comment (tianyizheng02) [inline greedy_methods/interval_scheduling.py:4]: Please add type hints

File: greedy_methods/interval_scheduling.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    # Initialize the result list with the first interval (earliest end time)
+    result = [sorted_intervals[0]]
+
+    for interval in sorted_intervals[1:]:
+        # Check if the start time of the current interval is after the end time of the last interval in the result
+        if interval[0] >= result[-1][1]:
+            result.append(interval)
+
+    return result
```
Comment (tianyizheng02) [inline greedy_methods/interval_scheduling.py:16]:
```suggestion
    # Sort intervals based on their end time
    sorted_intervals = sorted(intervals, key=lambda x: x[1])

    selected = [sorted_intervals[0]]
    for interval in sorted_intervals[1:]:
        # Add current interval if it doesn't overlap with the last selected interval
        if interval[0] >= result[-1][1]:
            selected.append(interval)

    return selected
```
1) Let's be less verbose in the comments, explain things at a relatively high level and avoid explaining every line of code
2) `result` isn't a very descriptive name, so I suggest changing it to something like `selected`

File: greedy_methods/interval_scheduling.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+# Example usage:
+# List of intervals represented as tuples (start_time, end_time)
+intervals = [(1, 4), (3, 6), (2, 8), (5, 9), (7, 10)]
+
+# Find the maximum non-overlapping intervals
+max_intervals = interval_scheduling(intervals)
+
+print("Maximum non-overlapping intervals:")
+print(max_intervals)
```
Comment (tianyizheng02) [inline greedy_methods/interval_scheduling.py:27]: Put demo code inside a `__main__` block.

File: greedy_methods/interval_scheduling.py (hunk 2 above)
Comment (tianyizheng02) [inline greedy_methods/interval_scheduling.py:4]: Please add doctests

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
This algorithm is actually called [Earliest Deadline First Scheduling](https://en.wikipedia.org/wiki/Earliest_deadline_first_scheduling), so please change the name of this PR, your file, and your function to reflect this.

We also already have a [scheduling/](https://github.com/TheAlgorithms/Python/tree/c9a7234a954dd280dc8192ae77a564e647d013d4/scheduling) directory, so please put this file in there.

**Comment by tianyizheng02** [comment]
@dhruvmanila This PR doesn't have any tests. How come the algorithms-keeper bot didn't point it out?

File: greedy_methods/interval_scheduling.py (hunk 2 above)
Comment (HN026) [reply greedy_methods/interval_scheduling.py:4]: sure I will make the changes and psh my PR again

File: greedy_methods/interval_scheduling.py (hunk 1 above)
Comment (cclauss) [reply greedy_methods/interval_scheduling.py:1]: If this is going to land, it should be with the other scheduling algorithms...
https://github.com/TheAlgorithms/Python/tree/master/scheduling

**Comment by tianyizheng02** [comment]
Closing due to inactivity. There are no type hints or doctests, and the demo code should be within a `__main__` block. This has been pointed out and is still unfixed.

**Pull request closed without merging**